src/AutoRest.CSharp/Common/AutoRest/Plugins/GeneratedCodeWorkspace.cs

```diff
@@ -253,6 +253,18 @@ public static GeneratedCodeWorkspace CreateExistingCodeProject(string outputDire
             return new GeneratedCodeWorkspace(project);
         }
 
+        public static async Task<CSharpCompilation> CreatePreviousContractFromDll(string xmlDocumentationpath, string dllPath)
+        {
+            var workspace = new AdhocWorkspace();
+            Project project = workspace.AddProject("PreviousContract", LanguageNames.CSharp);
+            project = project
+                .AddMetadataReferences(AssemblyMetadataReferences)
+                .WithCompilationOptions(new CSharpCompilationOptions(
+                    OutputKind.DynamicallyLinkedLibrary, nullableContextOptions: NullableContextOptions.Disable));
+            project = project.AddMetadataReference(MetadataReference.CreateFromFile(dllPath, documentation: XmlDocumentationProvider.CreateFromFile(xmlDocumentationpath)));
+            return await (new GeneratedCodeWorkspace(project)).GetCompilationAsync();
```

Comment:
we do not need to call `new GeneratedCodeWorkspace` here, actually the `GetCompilationAsync` just calls the `GetCompilationAsync` method on the project, therefore we could just do:
```suggestion
            return await project.GetCompilationAsync();
```
And do we really need properties only on `CSharpCompilation` instead of `Compilation`? Usually `Compilation` just works.

Reply:
Right

Reply:
did we get this resolved?

Reply:
forgot to change the return type, will update it.

Reply:
updated

Reply:
well my point is that we do not need to construct a `GeneratedCodeWorkspace`.
Just `await project.GetCompilationAsync()` could work as expected.